containers/unit_tests/TestWithoutInitializing.hpp

```diff
@@ -37,6 +37,16 @@
 #endif
 ///@}
 
+/// Some tests are skipped for unified memory space.
+///@{
+#if defined(KOKKOS_ENABLE_IMPL_CUDA_UNIFIED_MEMORY)
+#define GTEST_SKIP_IF_UNIFIED_MEMORY_SPACE \
+  GTEST_SKIP() << "skipping since unified memory requires additional fences";
```

Comment:
Shouldn't this only check for `CudaSpace`?
```suggestion
#define GTEST_SKIP_IF_UNIFIED_MEMORY_SPACE \
  if constexpr (std::is_same_v<typename TEST_EXECSPACE::memory_space, \ 
                               Kokkos::CudaSpace>)
    GTEST_SKIP() << "skipping since unified memory requires additional fences";
```
Where exactly do the extra fences come from?

Reply:
I can see that this fails when we fence consistently.

Reply:
ping